test/cloudnative/proxy.go

```diff
@@ -0,0 +1,151 @@
+		dynakube.NewBuilder().
+			WithDefaultObjectMeta().
+			ApiUrl(secretConfig.ApiUrl).
+			CloudNative(&v1beta1.CloudNativeFullStackSpec{}).
+			Proxy(proxySpec).
+			Build()),
+	)
+	setup.AssessDynakubeStartup(cloudNativeWithProxy)
+
+	cloudNativeWithProxy.Assess("restart sample apps", sampleapps.Restart)
+	cloudNativeWithProxy.Assess("check existing init container and env vars", checkSampleInitContainer)
+	cloudNativeWithProxy.Assess("check logs", checkLogs)
+
+	return cloudNativeWithProxy.Feature()
+}
+
+func checkLogs(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+	resources := environmentConfig.Client().Resources()
+	clientset, err := kubernetes.NewForConfig(resources.GetConfig())
+
+	require.NoError(t, err)
+
+	err = deployment.NewQuery(ctx, resources, client.ObjectKey{
+		Name:      "dynatrace-operator",
+		Namespace: "dynatrace",
+	}).ForEachPod(func(podItem v1.Pod) {
+		logStream, err := clientset.CoreV1().Pods(podItem.Namespace).GetLogs(podItem.Name, &v1.PodLogOptions{}).Stream(ctx)
+		require.NoError(t, err)
+		logs.AssertLogContains(t, logStream, timeoutError)
+
+		logStream, err = clientset.CoreV1().Pods(podItem.Namespace).GetLogs(podItem.Name, &v1.PodLogOptions{}).Stream(ctx)
+		require.NoError(t, err)
+		logs.AssertLogContains(t, logStream, secretUnchanged)
+	})
+
+	require.NoError(t, err)
+
+	err = deployment.NewQuery(ctx, resources, client.ObjectKey{
+		Name:      "myapp",
+		Namespace: "test-namespace-1",
+	}).ForEachPod(func(podItem v1.Pod) {
+		logStream, err := clientset.CoreV1().Pods(podItem.Namespace).GetLogs(podItem.Name, &v1.PodLogOptions{
+			Container: "install-oneagent",
+		}).Stream(ctx)
+		require.NoError(t, err)
+		logs.AssertLogContains(t, logStream, proxy.ProxySpec.Value)
+	})
+
+	require.NoError(t, err)
+	return ctx
+}
+
+func checkSampleInitContainer(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+	resources := environmentConfig.Client().Resources()
+	restConfig := environmentConfig.Client().RESTConfig()
+	pods := sampleapps.Get(t, ctx, resources)
+
+	for _, podItem := range pods.Items {
+		require.NotNil(t, podItem)
+		require.NotNil(t, podItem.Spec)
+		require.NotNil(t, podItem.Spec.InitContainers)
+
+		for _, container := range podItem.Spec.Containers {
+			if container.Name == "myapp" {
+				require.NotNil(t, container.Env)
+				require.True(t, containsDtProxyEnvVar(container.Env, DtProxy))
+				for _, env := range container.Env {
+					if env.Name == DtProxy {
+						require.NotNil(t, env.Value)
+					}
+				}
+			}
+		}
+
+		var result *pod.ExecutionResult
+		result, err := pod.
+			NewExecutionQuery(podItem, sampleapps.Name,
+				bash.ListDirectory(agentPath)).
+			Execute(restConfig)
+
+		require.NoError(t, err)
+		assert.Contains(t, result.StdOut.String(), "agent")
+		assert.Empty(t, result.StdErr.String())
+	}
+	return ctx
+}
+
+func containsDtProxyEnvVar(envs []v1.EnvVar, dtproxy string) bool {
```

Comment:
there should already exist a function that checks a set of env vars for a specific one in which case there is no need to reimplement it